Downstream build regression due to missing files [6478]

#738 broke downstream packages. Installed headers reference other headers which are missing.

On Linux:
```
In file included from src/ros2/rmw_fastrtps/rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_subscriber_info.hpp:24:0,
                 from src/ros2/rmw_fastrtps/rmw_fastrtps_shared_cpp/src/custom_subscriber_info.cpp:15:
install/fastrtps/include/fastrtps/subscriber/Subscriber.h:23:10: fatal error: fastdds/rtps/common/Guid.h: No such file or directory
 #include <fastdds/rtps/common/Guid.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

On Windows:
```
install\include\fastrtps/attributes/PublisherAttributes.h(22): fatal error C1083: Cannot open include file: 'fastdds/rtps/resources/ResourceManagement.h': No such file or directory (compiling source file src\ros2\rmw_fastrtps\rmw_fastrtps_shared_cpp\src\qos.cpp) [build\rmw_fastrtps_shared_cpp\rmw_fastrtps_shared_cpp.vcxproj]
```

Also reported here: https://github.com/eProsima/Fast-RTPS/pull/738#issuecomment-536602969

--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -380,6 +380,11 @@ set_public_headers_directory(${PROJECT_SOURCE_DIR}/include ${PROJECT_NAME}
     COMPONENT headers
     INSTALL
     )
+set_public_headers_directory(${PROJECT_SOURCE_DIR}/include fastdds
+    DESTINATION ${INCLUDE_INSTALL_DIR}
+    COMPONENT headers
+    INSTALL
+    )
 
 # Install config.h header
 set_public_header(${PROJECT_BINARY_DIR}/include ${PROJECT_NAME} config.h
@@ -406,6 +411,7 @@ set(CPACK_COMPONENT_HEADERS_DISPLAY_NAME "C++ Headers" PARENT_SCOPE)
 set(CPACK_COMPONENT_HEADERS_DESCRIPTION "eProsima ${PROJECT_NAME_LARGE} C++ Headers" PARENT_SCOPE)
 
 # Install sources
+
 if(UNIX AND EPROSIMA_INSTALLER)
     install(DIRECTORY ${PROJECT_SOURCE_DIR}/src/cpp
         DESTINATION src

